### PR TITLE
optional external blackbox exporter for probing shoots outside gardener

### PR DESCRIPTION
**Pull request by nschad**
**How to categorize this PR?**

/area monitoring
/kind enhancement
/kind api-change

**What this PR does / why we need it**:

Optional external blackbox exporter for probing shoots outside gardener

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Release note**:

```feature operator
* Add API blackbox exporter probe metrics from outside the gardener environment
```

### CODE REVIEW

**Comment by gardener-robot-ci-1** [comment]
Thank you @nschad for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

**Comment by wyb1** [comment]
I did not get a chance to take a look at this today. I will try to get to it by Wednesday.

File: pkg/operation/botanist/monitoring.go [hunk 1]
```diff
@@ -236,6 +236,37 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
 
 	prometheusConfig["podAnnotations"] = podAnnotations
 
+	// Add external blackbox exporter when enabled
+	if b.Config.Monitoring != nil &&
+		b.Config.Monitoring.Shoot != nil &&
+		b.Config.Monitoring.Shoot.ExternalBlackboxExporter != nil &&
+		b.Config.Monitoring.Shoot.ExternalBlackboxExporter.URL != "" {
+		// if remoteWrite Url is set add config into values
```
Comment (istvanballok) [inline pkg/operation/botanist/monitoring.go]: remoteWrite -> external blackbox exporter (probably copy paste leftover)

File: pkg/operation/botanist/monitoring.go [hunk 2]
```diff
@@ -236,6 +236,37 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
...
+		externalBlackboxExporterConfig := map[string]interface{}{
+			"url":    b.Config.Monitoring.Shoot.ExternalBlackboxExporter.URL,
+			"module": b.Config.Monitoring.Shoot.ExternalBlackboxExporter.Module,
+		}
+
+		for _, advertisedAddresse := range b.Shoot.GetInfo().Status.AdvertisedAddresses {
```
Comment (istvanballok) [inline pkg/operation/botanist/monitoring.go]: typo: advertisedAddress[e]

File: pkg/operation/garden/garden.go [hunk 3]
```diff
@@ -253,6 +253,12 @@ func ReadGardenSecrets(ctx context.Context, c client.Reader, namespace string, l
 			secretsMap[v1beta1constants.GardenRoleGlobalShootRemoteWriteMonitoring] = &monitoringSecret
 			logInfo = append(logInfo, fmt.Sprintf("monitoring basic auth secret %q", secret.Name))
 		}
+
+		if secret.Labels[v1beta1constants.GardenRole] == v1beta1constants.GardenRoleGlobalShootExternalBlackboxExporterMonitoring {
```
Comment (istvanballok) [inline pkg/operation/garden/garden.go:259]: maybe add a comment above this block to make it consistent with the other similar blocks

**Review by istvanballok — COMMENTED** [review]
Besides the minor comments, /lgtm

**Review by istvanballok — COMMENTED** [review]
/lgtm

File: pkg/operation/botanist/monitoring.go [hunk 4]
```diff
@@ -236,6 +236,37 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
...
+		externalBlackboxExporterConfig := map[string]interface{}{
+			"url":    b.Config.Monitoring.Shoot.ExternalBlackboxExporter.URL,
+			"module": b.Config.Monitoring.Shoot.ExternalBlackboxExporter.Module,
+		}
+
+		for _, advertisedAddress := range b.Shoot.GetInfo().Status.AdvertisedAddresses {
+			if advertisedAddress.Name == "external" {
+				externalBlackboxExporterConfig["externalApiUrl"] = advertisedAddress.URL
+			}
+		}
```
Comment (rfranzke) [inline pkg/operation/botanist/monitoring.go:254]: Use https://github.com/gardener/gardener/blob/bfe9f808f6af29949cd6e48edc4df02cc310bf97/pkg/operation/botanist/operatingsystemconfig.go#L91

**Review by rfranzke — COMMENTED** [review]
To be honest, I have some mixed feelings about additions like this one or also #5639. While I appreciate the contributions, it seems that the solutions are very specific and tailored to a specific use-case. Shouldn't such customizations better be put into an extension?

**Comment by nschad** [comment]
closing due to #5650

**Pull request closed without merging**